examples/Azure_IoT_Adu_ESP32/Azure_IoT_Adu_ESP32.ino

```diff
@@ -0,0 +1,1024 @@
+ * for client authentication);
+ * - Periodically send telemetry data to the Azure IoT Hub.
+ *
+ * To properly connect to your Azure IoT Hub, please fill the information in the
+ * `iot_configs.h` file.
+ */
+
+// C99 libraries
+#include <cstdlib>
+#include <string.h>
+#include <time.h>
+
+// Libraries for MQTT client and WiFi connection
+#include <HTTPClient.h>
+#include <HTTPUpdate.h>
+#include <WiFi.h>
+#include <mqtt_client.h>
+
+// Azure IoT SDK for C includes
+#include <az_core.h>
+#include <az_iot.h>
+#include <azure_ca.h>
+
+// Additional sample headers
+#include "AzIoTCryptoMbedTLS.h"
+#include "AzIoTSasToken.h"
+#include "SampleAduJWS.h"
+#include "SerialLogger.h"
+#include "iot_configs.h"
+
+#include "esp_ota_ops.h"
+#include "esp_system.h"
+
+// When developing for your own Arduino-based platform,
+// please follow the format '(ard;<platform>)'.
+#define AZURE_SDK_CLIENT_USER_AGENT "c%2f" AZ_SDK_VERSION_STRING "(ard;esp32)"
+
+#define SAMPLE_MQTT_TOPIC_LENGTH 128
+#define SAMPLE_MQTT_PAYLOAD_LENGTH 1024
+
+// ADU Values
+#define ADU_DEVICE_MANUFACTURER "ESPRESSIF"
+#define ADU_DEVICE_MODEL "ESP32-Embedded"
+#define ADU_DEVICE_VERSION "1.0"
+#define ADU_DEVICE_SHA_SIZE 32
+#define ADU_SHA_PARTITION_READ_BUFFER_SIZE 32
+#define HTTP_DOWNLOAD_CHUNK 4096
+
+// ADU Feature Values
+static az_iot_adu_client adu_client;
+static az_iot_adu_client_update_request adu_update_request;
+static az_iot_adu_client_update_manifest adu_update_manifest;
+static char adu_new_version[16];
+static bool process_update_request = false;
+static bool did_update = false;
+static char adu_scratch_buffer[10000];
+static char adu_verification_buffer[jwsSCRATCH_BUFFER_SIZE];
+static char adu_sha_buffer[ADU_DEVICE_SHA_SIZE];
+static char adu_calculated_sha_buffer[ADU_DEVICE_SHA_SIZE];
+static char partition_read_buffer[ADU_SHA_PARTITION_READ_BUFFER_SIZE];
+static int chunked_data_index;
+
+#define AZ_IOT_ADU_DTMI "dtmi:azure:iot:deviceUpdateModel;1"
+
+static az_span pnp_components[] = { AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_PROPERTIES_COMPONENT_NAME) };
+az_iot_adu_client_device_properties adu_device_information
+    = { .manufacturer = AZ_SPAN_LITERAL_FROM_STR(ADU_DEVICE_MANUFACTURER),
+        .model = AZ_SPAN_LITERAL_FROM_STR(ADU_DEVICE_MODEL),
+        .adu_version = AZ_SPAN_LITERAL_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_VERSION),
+        .delivery_optimization_agent_version = AZ_SPAN_EMPTY,
+        .update_id = { .provider = AZ_SPAN_LITERAL_FROM_STR(ADU_DEVICE_MANUFACTURER),
+                       .name = AZ_SPAN_LITERAL_FROM_STR(ADU_DEVICE_MODEL),
+                       .version = AZ_SPAN_LITERAL_FROM_STR(ADU_DEVICE_VERSION) } };
+
+// Utility macros and defines
+#define sizeofarray(a) (sizeof(a) / sizeof(a[0]))
+#define NTP_SERVERS "pool.ntp.org", "time.nist.gov"
+#define MQTT_QOS1 1
+#define DO_NOT_RETAIN_MSG 0
+#define SAS_TOKEN_DURATION_IN_MINUTES 60
+#define UNIX_TIME_NOV_13_2017 1510592825
+
+#define PST_TIME_ZONE -8
```

Comment:
Does this make the sample locked to PST?

Reply:
You know that's an interesting question. I know I'm running mine not in Pacific but we should make a note of this.